### PR TITLE
frontend/fix: #2698 Added Keyboard Listener

### LINKED ISSUE #2698: Chat related fixes
Driver side
1. If driver is on NY app, the current popup just mentions customer, change it to "From Namma Yatri Customer"

Rider side (Android)
1. If rider is clicking on "reply" inside push notification, the chat window of the app should open directly
2. in keyboard view of chat, the chat should be auto-scrolled until the latest message on the chat.

Rider side (Ios)
1. Any message from driver appears as a normal push notification. Change the logo for messaging related PNs to mascot + logo
2.  Also change the message header to "Message from driver"
3.  in keyboard view of chat, the chat should be auto-scrolled until the latest message on the chat.
4. The driver message should appear in lock screen state as well.

### PR DESCRIPTION
**Pull request by PraveenGongada**
## Type of Change

- [x] Bugfix
- [ ] New feature
- [ ] Enhancement
- [ ] Refactoring
- [ ] Dependency updates

## Description

- Fixed chat scroll view on keyBoardOpen
- changed title of In-App Notification

Closes #2698

### CODE REVIEW

**Review by Vignesh-772 — CHANGES_REQUESTED** [review]
Do the changes as discussed

**Review by Vignesh-772 — APPROVED** [review]

File: Frontend/android-native/mobility-app/src/main/java/in/juspay/mobility/app/ChatService.java [hunk 1]
```diff
@@ -305,6 +305,18 @@ else if(appName.equals("Yatri Sathi"))
         }
     }
 
+    private String getCustomerApp(String driverApp) {
+        switch (driverApp) {
+            case "Yatri Partner" :
```
Comment (MercyQueen) [inline Frontend/android-native/mobility-app/src/main/java/in/juspay/mobility/app/ChatService.java:310]: @PraveenGongada the app_name for Yatri partner is `Yatri Driver` .

File: Frontend/ui-driver/src/Screens/HomeScreen/Controller.purs [hunk 2]
```diff
@@ -310,6 +312,18 @@ eval TriggerMaps state = continueWithCmd state[ do
...
   pure NoAction
   ]
+
+eval (KeyboardCallback keyBoardState) state = do 
+  let isOpen = case keyBoardState of
+                    "onKeyboardOpen" -> true
+                    "onKeyboardClose" -> false
+                    _ -> false 
+  if state.props.currentStage == ST.ChatWithCustomer && isOpen then do
+    _ <- pure $ scrollToEnd (getNewIDWithTag "ChatScrollView") true
```
Comment (MercyQueen) [inline Frontend/ui-driver/src/Screens/HomeScreen/Controller.purs:322]: use void here and remove pure unit

File: Frontend/ui-driver/index.js [hunk 3]
```diff
@@ -221,14 +221,21 @@ window["onEvent'"] = function (event, args) {
...
-  // else if (event == "onTimeChanged") {
+  // } else if (event == "onTimeChanged") {
   //   if(window.dateCallback != undefined) {
   //     window.dateCallback();
   //   } else {
   //     purescript.onConnectivityEvent("CHECKING_DATE_TIME")();
-  //   }
-  // }   -- Need To Refactor 
+  //   }  -- Need To Refactor 
+  } else if (event == "onKeyboardOpen") {
```
Comment (MercyQueen) [inline Frontend/ui-driver/index.js:230]: use `||` operator to combine the two else if statements

File: Frontend/ui-customer/src/Screens/RideBookingFlow/HomeScreen/Controller.purs [hunk 4]
```diff
@@ -635,6 +636,17 @@ eval TerminateApp state = do
   pure $ terminateApp state.props.currentStage true
   continue state
 
+eval (KeyboardCallback keyBoardState) state = do 
+  let isOpen = case keyBoardState of
+                    "onKeyboardOpen" -> true
+                    "onKeyboardClose" -> false
+                    _ -> false 
+  if isLocalStageOn ChatWithDriver then do 
+    _ <- pure $ scrollToEnd (getNewIDWithTag "ChatScrollView") true
```
Comment (MercyQueen) [inline Frontend/ui-customer/src/Screens/RideBookingFlow/HomeScreen/Controller.purs:645]: use void here as well.

File: Frontend/ui-customer/index.js [hunk 5]
```diff
@@ -228,6 +228,14 @@ window["onEvent'"] = function (event, args) {
     purescript.onConnectivityEvent("LOCATION_DISABLED")();
   } else if (event == "onInternetChanged") {
     purescript.onConnectivityEvent("INTERNET_ACTION")();
+  } else if (event == "onKeyboardOpen") {
```
Comment (MercyQueen) [inline Frontend/ui-customer/index.js:231]: @PraveenGongada here also combine if else statements using `||` operator.

**Review by MercyQueen — CHANGES_REQUESTED** [review]

**Review by MercyQueen — DISMISSED** [review]

File: Frontend/android-native/mobility-common/src/main/java/in/juspay/mobility/common/MobilityCommonBridge.java [hunk 6]
```diff
@@ -257,6 +262,15 @@ public void reset() {
         markers = new JSONObject();
         pickupPointsZoneMarkers = new ArrayList<>();
         layer = null;
+        if(onGlobalLayoutListener != null && bridgeComponents.getActivity() != null) {
```
Comment (Rohit4997) [inline Frontend/android-native/mobility-common/src/main/java/in/juspay/mobility/common/MobilityCommonBridge.java:265]: This condition can be sinplified

File: Frontend/android-native/mobility-common/src/main/java/in/juspay/mobility/common/MobilityCommonBridge.java [hunk 7]
```diff
@@ -431,6 +445,49 @@ private void getLastKnownLocationFromClientFallback(String callback, boolean ani
...
+                        int estimatedKeyboardHeight = (int) TypedValue.applyDimension(TypedValue.COMPLEX_UNIT_DIP, estimatedKeyboardDP, parentView.getResources().getDisplayMetrics());
+                        parentView.getWindowVisibleDisplayFrame(rect);
+                        int heightDiff = parentView.getRootView().getHeight() - (rect.bottom - rect.top);
+                        boolean isShown = heightDiff >= estimatedKeyboardHeight;
+
+                        if (isShown == alreadyOpen) {
+                            return;
+                        }
+                        alreadyOpen = isShown;
+                        onKeyboardVisibilityListener.onVisibilityChanged(isShown);
```
Comment (Rohit4997) [inline Frontend/android-native/mobility-common/src/main/java/in/juspay/mobility/common/MobilityCommonBridge.java:483]: Can't we call simple callback here?
`                        String encoded = Base64.encodeToString("{}".getBytes(), Base64.NO_WRAP);
                        String command = String.format("window[\"onEvent'\"]('%s',atob('%s'))", visible ? "onKeyboardOpen" : "onKeyboardClose", encoded);
                        bridgeComponents.getJsCallback().addJsToWebView(command);`
                        
 What is the use of interface ?

File: Frontend/ui-common/src/Engineering/Helpers/JBridge.js [hunk 8]
```diff
@@ -650,6 +650,17 @@ export const storeCallBackMessageUpdated = function (cb) {
   };
 };
 
+export const storeKeyBoardCallback = function (cb) {
```
Comment (Rohit4997) [inline Frontend/ui-common/src/Engineering/Helpers/JBridge.js:653]: Used uncurried function

File: Frontend/ui-common/src/Engineering/Helpers/JBridge.purs [hunk 9]
```diff
@@ -229,6 +229,7 @@ foreign import getLayoutBounds :: Fn1 String LayoutBound
 foreign import horizontalScrollToPos :: EffectFn3 String String Int Unit
 foreign import withinTimeRange :: String -> String -> String -> Boolean
 foreign import getChatMessages :: String -> Array ChatComponent
+foreign import storeKeyBoardCallback :: forall action. (action -> Effect Unit) -> (String -> action) -> Effect Unit
```
Comment (Rohit4997) [inline Frontend/ui-common/src/Engineering/Helpers/JBridge.purs:232]: Use uncurried function

File: Frontend/ui-customer/index.js [hunk 10]
```diff
@@ -228,6 +228,10 @@ window["onEvent'"] = function (event, args) {
     purescript.onConnectivityEvent("LOCATION_DISABLED")();
   } else if (event == "onInternetChanged") {
     purescript.onConnectivityEvent("INTERNET_ACTION")();
+  } else if (event == "onKeyboardOpen" || event == "onKeyboardClose") {
```
Comment (Rohit4997) [inline Frontend/ui-customer/index.js:231]: `else if ((event == "onKeyboardOpen" || event == "onKeyboardClose") && window.keyBoardCallback)`
Simplify this

**Review by Rohit4997 — CHANGES_REQUESTED** [review]

**Review by Rohit4997 — DISMISSED** [review]

**Review by Vignesh-772 — DISMISSED** [review]

**Pull request merged**